client/packages/programs/src/JsonForms/common/components/Array/EnumArray.tsx

```diff
@@ -76,12 +85,25 @@ export const EnumArrayComponent: FC<EnumArrayControlCustomProps> = ({
   path,
   schema,
   visible,
+  enabled,
   addItem,
   removeItems,
 }) => {
   const t = useTranslation('common');
   const [removeIndex, setRemoveIndex] = useState<number | undefined>();
 
+  const renderInput = (props: AutocompleteRenderInputParams) => (
```

Comment:
can you move this outside of the `EnumArrayComponent` please?
Not really an issue in this case, as the component is not large; it's a good habit to have tho.
Currently the `renderInput` component is recreated on every render, which means that if `EnumArrayComponent` has to re-render, it redraws all child components. If the `renderInput` is created at a higher level, then only the updated elements need to redraw.

Reply:
done thanks!